cmd/skaffold/app/cmd/deploy.go

```diff
@@ -43,31 +43,35 @@ func NewCmdDeploy() *cobra.Command {
 		WithExample("Build the artifacts and collect the tags into a file", "build --file-output=tags.json").
 		WithExample("Deploy those tags", "deploy --build-artifacts=tags.json").
 		WithExample("Build the artifacts and then deploy them", "build -q | skaffold deploy --build-artifacts -").
+		WithExample("Deploy without first rendering the manifests", "skaffold deploy --skip-render").
 		WithCommonFlags().
 		WithFlags(func(f *pflag.FlagSet) {
 			f.VarP(&preBuiltImages, "images", "i", "A list of pre-built images to deploy")
 			f.VarP(&buildOutputFile, "build-artifacts", "a", `Filepath containing build output.
 E.g. build.out created by running skaffold build --quiet -o "{{json .}}" > build.out`)
+			f.BoolVar(&opts.SkipRender, "skip-render", false, "Don't render the manifests, just deploy them")
 		}).
 		NoArgs(doDeploy)
 }
 
 func doDeploy(ctx context.Context, out io.Writer) error {
 	return withRunner(ctx, func(r runner.Runner, config *latest.SkaffoldConfig) error {
-		deployed, err := getArtifactsToDeploy(out, buildOutputFile.BuildArtifacts(), preBuiltImages.Artifacts(), config.Build.Artifacts)
-		if err != nil {
-			return err
-		}
-
-		for i := range deployed {
-			tag, err := r.ApplyDefaultRepo(deployed[i].Tag)
+		var deployed []build.Artifact
+		if !opts.SkipRender {
```

Comment:
you can simplify this by return first when `--skip-render=true`
```suggestion
		if opts.SkipRender {
		   return r.DeployAndLog(ctx, out, deployed)
		}
```

Reply:
as far as i understand, with `--skip-render=true` deploy should happen. However with your change, seems like deploy is skipped all together.
Am i reading this correct?

Reply:
With your suggestion, `return r.DeployAndLog(ctx, out, deployed)` will show up twice. Should I duplicate that line of code?

deploy is not skipped as `return r.DeployAndLog(ctx, out, deployed)` is the code to do the deployment.

Reply:
yes. however, `deployed` is empty in `opts.SkipRender=false` flow. So no artifacts will be deployed?
I will try to run this locally today.

Reply:
ah got. The `--skip-render` does not replace the built images and deploys the manifests as is.

Reply:
>With your suggestion, return r.DeployAndLog(ctx, out, deployed) will show up twice. Should I duplicate that line of code?

Yes, i think it will easier to read if you add the `opts.SkipRender` check as first line of the function and return and pass an empty build.Artifact array.
```
return r.DeployAndLog(ctx, out, []build.Artifact{})
``